File: Lab 4/Lab 4 c++/Desenvolvedor.cpp

#include "Desenvolvedor.h"

Desenvolvedor::Desenvolvedor(const std::string& nome, int salario, const std::string& linguagemPrincipal)
    : Funcionario(nome, salario), linguagemPrincipal(linguagemPrincipal) {}

std::string Desenvolvedor::getLinguagemPrincipal() const {
    return linguagemPrincipal;
}

void Desenvolvedor::setLinguagemPrincipal(const std::string& linguagemPrincipal) {
    this->linguagemPrincipal = linguagemPrincipal;
}

std::string Desenvolvedor::projetosRealizados() const {
    return "\nProjetos entregues na linguagem " + linguagemPrincipal;
}

std::string Desenvolvedor::descricaoDesenvolvedor() const {
    return "\nLinguagem do desenvolvedor: " + linguagemPrincipal;
}

std::string Desenvolvedor::descricao() const {
    return Funcionario::descricao() + descricaoDesenvolvedor();
}

File: Lab 4/Lab 4 Java/Desenvolvedor.java

public class Desenvolvedor extends Funcionario {
    private String linguagemPrincipal;

    public Desenvolvedor(String nome, int salario, String linguagemPrincipal) {
        super(nome, salario);
        this.linguagemPrincipal = linguagemPrincipal;
    }

    @Override
    public String descricao() {
        return super.descricao() + "\nLinguagem Principal: " + linguagemPrincipal;
    }
}
